### ISSUE TITLE
Dark mode

### ISSUE THREAD

**Issue by george-emerald** [erengy/taiga#807]
This is an amazing app for tracking and organizing anime and I'm very thankful to great open source programs that people spend their free time to make without pay. But, I just don't understand why this feature has been postponed for so long. It has been requested very many times for years now and each time it's pushed to a v. 2.0 that doesn't even have a branch here yet. The High Contrast Theme solution that has been posited completely destroys your entire Windows UI. Is it really this difficult to implement a button that changes the colours from white to black and a shade of dark grey? The blinding white background is only present in this one program in my entire computer.

**Comment by erengy** [comment]
> Is it really this difficult to implement a button that changes the colours from white to black and a shade of dark grey?

You must understand that Taiga doesn't paint its UI, Windows does. All the controls you see are almost entirely drawn and painted by the operating system. This is because Taiga is a native desktop application, which is kind of old-school by today's standards. It makes Taiga efficient and consistent, but it also makes some features unfeasible.

Changing to dark mode is not as simple as saying, "hey Windows, paint my application with this palette, please." I wish it was. It's more like, "give me the tools, I'll paint my own goddamn application myself." Not only that, but some tools are unreliable, some stuff are practically unpaintable, and the guides are written in Ancient Greek.

I tried. It took me more than 20 hours just to get the main window and the anime list *partially* dark. This is a great deal of work that really should be handled by the OS. Microsoft seems to be slowly bringing the dark mode to desktop apps such as File Explorer, so hopefully we'll see more of it in future updates.

**Comment by 0ANIMEFREAK0** [comment]
@george-emerald if u don't mind using theme u can get pure dark mode by installing this [theme](https://www.deviantart.com/niivu/art/Blossom-Windows-10-Suite-808994246)

![image](https://user-images.githubusercontent.com/35454710/63800541-2ea9bd80-c930-11e9-856a-283e0aff55b3.png)

**Comment by avail** [comment]
https://stackoverflow.com/a/55670413
you can use this or similar to retrieve windows colours and have them auto-swap depending on theme

**Comment by george-emerald** [comment]
> https://stackoverflow.com/a/55670413
> you can use this or similar to retrieve windows colours and have them auto-swap depending on theme

The thread says that you can somehow make Win32 applications built on C++ change their colour depending on your theme, but I can't figure out how to do it. Do you change some registry value? Input lines of code on Powershell or cmd? Please explain how to enable this as it could fix the problem.

**Comment by erengy** [comment]
I wish, but there isn't a hidden trick like that. Changing that registry value is the same as changing the color from *Settings → Personalization → Colors*, which doesn't affect Win32 applications. The accepted answer simply provides a way to get the setting programmatically.

This, or retrieving the colors, is like 1% of what's currently necessary to make Taiga dark themed. Sadly, my previous comment still applies. And yes, I'm aware of undocumented APIs such as the ones in ysc3839/win32-darkmode.

In short, Taiga won't have dark mode until either Microsoft officially allows that (maybe after WinUI 3 release this year?), or until after v2 where Taiga will switch to a framework such as Qt or wxWidgets.

**Comment by george-emerald** [comment]
> This, or retrieving the colors, is like 1% of what's currently necessary to make Taiga dark themed. Sadly, my previous comment still applies. And yes, I'm aware of undocumented APIs such as the ones in ysc3839/win32-darkmode.

How can I use the undocumented API?

**Comment by beep4beep** [comment]
The dark theme on windows 10 has evolved a little, nearly every app is dark themed now, including my windows file explorer. 

Remember the advanced option "Application UI Theme"? You can add your own take on the taiga theme by making a copy from the default theme folder, then edit the xml.  Once you are done you can change default to the name of the folder of your new theme. Sadly it only works for the progress bar.  Meaning the background for the graph etc. 

Really hoping for a better solution.

**Comment by DaCat1337** [comment]
Is there an update or perhaps a workaround for this issue ?

**Comment by lasersPew** [comment]
On that note, why not use [Qt](https://en.wikipedia.org/wiki/Qt_(software))?

**Comment by erengy** [comment]
v2 now supports dark mode out of the box. It's not final, but currently looks like this:

![image](https://github.com/user-attachments/assets/dd38fae8-cf20-4299-8e82-6a1c3707ce9f)

**Comment by AndreiAdii** [comment]
> v2 now supports dark mode out of the box. It's not final, but currently looks like this:
> 
> ![image](https://private-user-images.githubusercontent.com/1078430/380437952-dd38fae8-cf20-4299-8e82-6a1c3707ce9f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3ODAzMjQsIm5iZiI6MTczMDc4MDAyNCwicGF0aCI6Ii8xMDc4NDMwLzM4MDQzNzk1Mi1kZDM4ZmFlOC1jZjIwLTQyOTktOGU4Mi02YTFjMzcwN2NlOWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEwNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMDVUMDQxMzQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzY4YThhOTA3OGYyZWY2YjEwNTBhNmUyNmY1Nzg2ZTZhMjNiODU3MjMwOWI2YzUzZmQxOGI1NjZmYWQ5NjYxMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.y1qm1p6Sr-tEZzv-YuS-VfYHipg-zBTCidH8ZsLysY4)

Is there a nightly release anywhere? I would really, really love to use it

**Comment by erengy** [comment]
It's just the UI right now, most functionality is missing. I'd love to make a release and gather some feedback, but it's not at that stage yet.

**Comment by george-emerald** [comment]
Are there any plans to make v2 cross-platform?

**Comment by Altarus94** [comment]
For anyone still hoping for a dark Taiga on v1: I put together an **unofficial dark-mode build of 1.4.1**. It darkens the whole UI — anime list, sidebar, menus, toolbars, dialogs, Seasons, Statistics, Now Playing — not just the progress bars. It auto-enables on Windows 10 1809+ and falls back to the normal light UI otherwise (and under High Contrast). Everything else works exactly like stock Taiga; it reads your existing list/settings.

Download + source: https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark

Not affiliated with the project and not trying to step on v2 — just sharing in case it's useful to people here. GPL-3.0 like upstream.

**Comment by Smeagol001** [comment]
> For anyone still hoping for a dark Taiga on v1: I put together an **unofficial dark-mode build of 1.4.1**. It darkens the whole UI — anime list, sidebar, menus, toolbars, dialogs, Seasons, Statistics, Now Playing — not just the progress bars. It auto-enables on Windows 10 1809+ and falls back to the normal light UI otherwise (and under High Contrast). Everything else works exactly like stock Taiga; it reads your existing list/settings.
> 
> Download + source: https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark
> 
> Not affiliated with the project and not trying to step on v2 — just sharing in case it's useful to people here. GPL-3.0 like upstream.

one little oversight  the text in the now playing tab https://prnt.sc/KU7Y23dLTgk3 is black on black atleast for me on win11 darkmode enabled systemwide
same with the streaming media tab in the options there its white on white https://prnt.sc/WSoI6aIYvO6U

**Comment by Altarus94** [comment]
@Smeagol001 thanks, great catch — both turned out to be **Windows 11-specific** (I could only test on Windows 10 dark mode, where both pages render fine, so I was blind to these).

Cause:
- **Now Playing black-on-black:** the anime info page returned a dark background brush but relied on the *default* text color. Windows 10 dark mode makes that light, but Windows 11 leaves it black. Now set explicitly.
- **Streaming providers white-on-white:** that list view's row background wasn't being forced dark. Added a fallback that forces dark rows + light text on plain list views.

I've pushed an updated build to the **same link** — just re-download `Taiga.exe`:
https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark

Since I don't have a Windows 11 machine to test on, could you give it a try and let me know if those two pages look right now? Really appreciate the detailed report + screenshots.

**Comment by Smeagol001** [comment]
> [@Smeagol001](https://github.com/Smeagol001) thanks, great catch — both turned out to be **Windows 11-specific** (I could only test on Windows 10 dark mode, where both pages render fine, so I was blind to these).
> 
> Cause:
> 
> * **Now Playing black-on-black:** the anime info page returned a dark background brush but relied on the _default_ text color. Windows 10 dark mode makes that light, but Windows 11 leaves it black. Now set explicitly.
> * **Streaming providers white-on-white:** that list view's row background wasn't being forced dark. Added a fallback that forces dark rows + light text on plain list views.
> 
> I've pushed an updated build to the **same link** — just re-download `Taiga.exe`: https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark
> 
> Since I don't have a Windows 11 machine to test on, could you give it a try and let me know if those two pages look right now? Really appreciate the detailed report + screenshots.

sure allmost perfect 
now streamig media  still shows abit of white https://prnt.sc/YNKPRyG4faD1 but tbh i could live with that :P oh i realised that it only shows that way when streaming media recognision is turned off
aparantly same for mediaplayers if i enable both categories and uncheck an option its fine and doesnt change that wierd way as when i have the whole sectrion disabled
soo that made me click every checkbox i could find to see if smh simmilar will happen and.... it doesnt :)
and the white here is intended ig? https://prnt.sc/U7Rmn3J57n8n would be nicer if it were a shade of gray i think somehow my eyes now  instantly focus on the bars
i think thebest would be  same colour than bg of the row but a hue brighter so yu see where the inteded bar is ending?
oh just  fond another  small thing  when you hover over the episode count it turns like this https://prnt.sc/xbjTmZInMzsx when i fling the mouse over the episodecount 
this looks kinda  jaring too https://prnt.sc/uwsAKhyejif6

**Comment by Altarus94** [comment]
@Smeagol001 thanks for the thorough second pass — all of it is fixed in the updated build (same link, just re-download `Taiga.exe`): https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark

- **List going light when a section is disabled:** great detective work on the enable/disable pattern — that was exactly it. Windows repaints *disabled* list views with light system colors and ignores every dark override, so those lists are no longer truly disabled in dark mode: they stay themed (dark scrollbars included) and get dimmed text instead.
- **Bright progress bars:** implemented pretty much what you suggested — when dark mode is on and the theme's track color is light (the stock theme), the track/border/buttons switch to a track slightly brighter than the row background. Status fill colors stay as the theme defines them.
- **Garbled episode counter on hover:** two bugs, actually. The stock theme's +/- hover buttons were light-UI colors (fixed by the same adaptation), and the hover chrome was never erased from rows when the mouse moved away quickly — invisible on a white background for years, glaring on dark. Rows now repaint when they lose hover.
- **The edit page (your last screenshot):** the date pickers and spinner arrows have no dark variant in Windows at all, so they're now custom-drawn dark. The drop-down month calendar is dark too as of the latest build.

Keep them coming if you spot anything else.

**Comment by Smeagol001** [comment]
> [@Smeagol001](https://github.com/Smeagol001) thanks for the thorough second pass — all of it is fixed in the updated build (same link, just re-download `Taiga.exe`): https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark
> 
> * **List going light when a section is disabled:** great detective work on the enable/disable pattern — that was exactly it. Windows repaints _disabled_ list views with light system colors and ignores every dark override, so those lists are no longer truly disabled in dark mode: they stay themed (dark scrollbars included) and get dimmed text instead.
> * **Bright progress bars:** implemented pretty much what you suggested — when dark mode is on and the theme's track color is light (the stock theme), the track/border/buttons switch to a track slightly brighter than the row background. Status fill colors stay as the theme defines them.
> * **Garbled episode counter on hover:** two bugs, actually. The stock theme's +/- hover buttons were light-UI colors (fixed by the same adaptation), and the hover chrome was never erased from rows when the mouse moved away quickly — invisible on a white background for years, glaring on dark. Rows now repaint when they lose hover.
> * **The edit page (your last screenshot):** the date pickers and spinner arrows have no dark variant in Windows at all, so they're now custom-drawn dark. The drop-down month calendar is dark too as of the latest build.
> 
> Keep them coming if you spot anything else.

hmm the hover over  epcount thing is till there but only if you be over it for a longer time now.
ig most users wont ever hover there for half a sec or so
i was now just deliberately staying there for a longer duration rather than flicking over it when i navigate
i mean youd usually just accidentally flick over it either way since  that hover thing seems to have no function a tall right? why does it even change?

**Comment by Altarus94** [comment]
@Smeagol001 an update on the hover thing.

I couldn't reproduce exactly what you were describing by hovering the episode count — and from what I can see, it renders correctly now.

**While digging into it I did find a real bug that matches your symptoms — and it turns out it's a latent bug in upstream Taiga**, not something dark-specific. Drawing the little ▼ arrow in the score box switches the shared drawing context to a size-5 font and never restores it, so any text repainted afterwards (the episode counters) inherits the micro font and turns into unreadable squiggles. The longer the hover, the more repaints get contaminated — which fits what you described. It's been there for years in the light UI too, just far less noticeable. I found and fixed four font leaks of the same kind.

Updated build on the same link: https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark — if the counters still garble for you on this build, a screenshot would help.

**Comment by Smeagol001** [comment]
> [@Smeagol001](https://github.com/Smeagol001) an update on the hover thing.
> 
> I couldn't reproduce exactly what you were describing by hovering the episode count — and from what I can see, it renders correctly now.
> 
> **While digging into it I did find a real bug that matches your symptoms — and it turns out it's a latent bug in upstream Taiga**, not something dark-specific. Drawing the little ▼ arrow in the score box switches the shared drawing context to a size-5 font and never restores it, so any text repainted afterwards (the episode counters) inherits the micro font and turns into unreadable squiggles. The longer the hover, the more repaints get contaminated — which fits what you described. It's been there for years in the light UI too, just far less noticeable. I found and fixed four font leaks of the same kind.
> 
> Updated build on the same link: https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark — if the counters still garble for you on this build, a screenshot would help.

https://prnt.sc/WpliI3IoRksp its still there but honestly its just a really minor thing since you actually have to hover over em and wait abit  on the counter itself for it to happen
oh and i just noticed it only  happens on the white parts  https://prnt.sc/IBs4PS2U2vjP

**Comment by Altarus94** [comment]
Found it, and your "only on the white parts" hint was what cracked it.

The counter is drawn in three pieces: the separator gets an explicit color, the two numbers don't — they just inherit whatever text color is already in the device context. On a hot-tracked row the list leaves the theme's hot-state color there, so the numbers ended up drawn in a near-background color while the "/" stayed readable. That's why only parts of it broke, and only while hovering. Another one that's been in Taiga all along, invisible on the light UI.

Fixed and uploaded to the same link: https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark

Tested with both the stock theme and a custom one, hovering the counter for several seconds. Thanks for sticking with it — you found three real bugs in this thing.

**Comment by Janzzze** [comment]
Really loving this dark mode version, did spot one issue the sizing handle in the bottom right corner seems to be broken and shows an error symbol instead

<img width="111" height="159" alt="Image" src="https://github.com/user-attachments/assets/c77535c8-0681-44f8-8405-acc91cb4ed5f" />

**Comment by Altarus94** [comment]
@Janzzze thanks, glad you like it. Good catch on the corner.

When nothing is pending, the status bar hides its update timer and cancel button by giving them zero width. The dark status bar drawing still painted their icons, so the red cancel icon landed in the corner, and the grip itself wasn't drawn at all. Resizing from the corner still worked, only the drawing was wrong.

Fixed in the build on the same link: https://github.com/Altarus94/taiga/releases/tag/v1.4.1-dark

**Comment by Janzzze** [comment]
found a couple more bugs
on the torrents page when clicking "check for new torrents" or when using search, the background flashes white
on the seasons page when click "refresh data" the back ground goes white until the new data is loaded
on the statistics page the scroll bar is still using the light theme

https://github.com/user-attachments/assets/8c4b813b-0605-4d5e-84cc-4b029f56cc5b
